Program.cs

```diff
@@ -1,63 +1,86 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using Microsoft.Azure.Management.Fluent;
-using Microsoft.Azure.Management.ResourceManager.Fluent;
-using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
-using Microsoft.Azure.Management.ResourceManager.Fluent.Models;
+using Azure;
+using Azure.Core;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
 using Microsoft.Azure.Management.Samples.Common;
-using Newtonsoft.Json.Linq;
-using System;
 
 namespace DeployUsingARMTemplate
 {
     public class Program
     {
+        private static ResourceIdentifier? _resourceGroupId = null;
+
         /**
          * Azure Resource sample for deploying resources using an ARM template.
          */
-        public static void RunSample(IAzure azure)
+        public static async Task RunSample(ArmClient client)
         {
-            var rgName = SdkContext.RandomResourceName("rgRSAT", 24);
-            var deploymentName = SdkContext.RandomResourceName("dpRSAT", 24);
-
+            var rgName = "rgRSAT";

```

Comment:
Let's add comments here to tell our customers, if they want to direct use this sample, which part they should replace with their own value, which part they can keep untouched, clarity the things customers need to do.